From 31cf5d1e5cab414e2785e829f4e70a5bd0f4ed38 Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Wed, 6 Mar 2024 16:36:31 +0100
Subject: [PATCH 0963/1141] counter: Introduce the COUNTER_COMP_FREQUENCY()
 macro

Now that there are two users for the "frequency" extension, introduce a
new COUNTER_COMP_FREQUENCY() macro.
This extension is intended to be a read-only signal attribute.

Suggested-by: William Breathitt Gray <william.gray@linaro.org>
Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Link: https://lore.kernel.org/r/20240306153631.4051115-1-fabrice.gasnier@foss.st.com
Signed-off-by: William Breathitt Gray <william.gray@linaro.org>
(cherry picked from commit f7c7d5fcbf199fa7f4dd98cfbeaca83cf509a760)
Change-Id: Ic519dde6a4388cfb003fabe7e5d4f02c9fbc4b97
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/365810
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 include/linux/counter.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/linux/counter.h b/include/linux/counter.h
index b63746637de2..9e378c98bc87 100644
--- a/include/linux/counter.h
+++ b/include/linux/counter.h
@@ -602,6 +602,9 @@ struct counter_array {
 #define COUNTER_COMP_FLOOR(_read, _write) \
 	COUNTER_COMP_COUNT_U64("floor", _read, _write)
 
+#define COUNTER_COMP_FREQUENCY(_read) \
+	COUNTER_COMP_SIGNAL_U64("frequency", _read, NULL)
+
 #define COUNTER_COMP_POLARITY(_read, _write, _available) \
 { \
 	.type = COUNTER_COMP_SIGNAL_POLARITY, \
-- 
2.39.2

